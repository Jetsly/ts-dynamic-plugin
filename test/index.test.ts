import * as ts from 'typescript';
import { join } from 'path';
import { readdirSync, readFileSync, statSync } from 'fs';
import transformerFactory from '../src/index';
const printer = ts.createPrinter();
const FIXTURES_DIR = join(__dirname, '__fixtures__');

const transformer = transformerFactory({
  funcName:'formatLocale',
  importDecla:`import { formatLocale } from './utils/locale'` 
})

describe('ts-dynamic-plugin', () => {
  readdirSync(FIXTURES_DIR).forEach(fixtureName => {
    const fixtureFile = join(FIXTURES_DIR, fixtureName);
    if (statSync(fixtureFile).isFile()) {
      it(fixtureName.split('-').join(' '), () => {
        const sourceCode = readFileSync(fixtureFile, 'utf-8');
        const source = ts.createSourceFile(fixtureName, sourceCode, ts.ScriptTarget.ES2016, true);
        const result = ts.transform(source, [transformer]);
        const transformedSourceFile = result.transformed[0];
        const resultCode = printer.printFile(transformedSourceFile);
        expect(resultCode).toMatchSnapshot();
      });
    }
  });
});