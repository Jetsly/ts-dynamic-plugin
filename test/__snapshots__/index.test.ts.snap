// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-dynamic-plugin auto import.ts 1`] = `
"import { formatLocale } from './utils/locale';
const a = formatLocale({ id: 'string' });
"
`;
